### PR TITLE
React: Added Ambient Module Declaration alongside Ambient External Module Declaration

### PR DESCRIPTION
**Pull request by bgrieder**
Make ambient module declarations consistent with other major modules (JQuery, lodash, etc…)
Having an Ambient Module Declaration of React makes React easier to extend

### CODE REVIEW

**Comment by vsiao** [comment]
We have https://github.com/borisyankov/DefinitelyTyped/blob/master/react/react-global.d.ts for this. See the README at https://github.com/borisyankov/DefinitelyTyped/tree/master/react and https://github.com/borisyankov/DefinitelyTyped/pull/3615 for relevant discussion.

**Comment by 1two** [comment]
My apologies, I did not see the discussion before opening this PR

I understand the issue can only be fixed by the Typescript team reworking the syntax of declaration files:  allowing external modules declarations to be created by importing internal modules without exposing the internal module name.

Yet, I am not sure I support the decision to *not* go with the crowd by splitting the definitions in two files in the meantime. This adds to the general confusion between libraries definitions and makes things a lot harder (including for our IDE) just to allow writing `import react = require("react")` **and** not "polluting" the global namespace with `React`. I cannot believe that people having this requirement have it with other, even more popular libraries that do not split files and that they have not adopted a general attitude towards their imports (or just do what seems to be a rather simple refactoring)

In summary, I believe this is a flaw in the definition files that must be fixed by the Typescript team, and not by implementing non recommended temporary workarounds like this one.

Yet, feel free to close and disregard this PR

**Comment by 1two** [comment]
As a matter of fact IntelliJ 14.1 auto-completion works much better with two separate files... (at least the scenario of the external module definition with CommonJS modules)
Since IJ is our IDE of choice and tooling is of primary importance for the team, I closed this PR.
